### PR TITLE
Refactor argument and entry point types

### PR DESCRIPTION
**Pull request by achocron**

### CODE REVIEW

File: src/QsCompiler/BondSchemas/Argument/Argument.bond [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-namespace Microsoft.Quantum.QsCompiler.BondSchemas.EntryPoint;
```
Comment (cesarzc) [inline src/QsCompiler/BondSchemas/Argument/Argument.bond:4]: I think it makes sense to have all the schemas in the same namespace. Having said that, maybe something like `Microsoft.Quantum.QsCompiler.BondSchemas.Execution`?

File: src/QsCompiler/Compiler/Compiler.csproj [hunk 2]
```diff
@@ -31,12 +31,4 @@
   <ItemGroup>
     <Service Include="{508349b6-6b84-4df5-91f0-309beebad82d}" />
   </ItemGroup>
-
-  <ItemGroup>
```
Comment (achocron) [inline src/QsCompiler/Compiler/Compiler.csproj:35]: Restore this

File: src/QsCompiler/Tests.Compiler/EntryPointGenerationTests.fs [hunk 3]
```diff
@@ -7,49 +7,32 @@ open System.IO
...
 open Microsoft.Quantum.QsCompiler
-open Microsoft.Quantum.QsCompiler.BondSchemas.EntryPoint
 open System.Text
 
 type EntryPointGenerationTests(output: ITestOutputHelper) =
 
     let testCasesDirectory = Path.Combine("TestCases", "QirEntryPointTests")
 
     [<Theory>]
-    [<InlineData("UseNoArgs")>]
```
Comment (achocron) [inline src/QsCompiler/Tests.Compiler/EntryPointGenerationTests.fs:18]: Re-add this test

File: src/QsCompiler/BondSchemas/Execution/Execution.cs [hunk 4]
```diff
@@ -83,43 +83,38 @@ public enum ResultValue
 
     [global::Bond.Schema]
     [System.CodeDom.Compiler.GeneratedCode("gbc", "0.12.1.0")]
-    public partial class ArrayValue
+    public partial class Argument
```
Comment (ScottCarda-MS) [inline src/QsCompiler/BondSchemas/Execution/Execution.cs:86]: Without values, and with its use in the Entry Point Operations, could we rename this type to `Parameter` instead? The term `argument` is more of a value that is bound to a parameter at run time, where as the term `parameter` is more related to the definition of the callable.

File: src/QsCompiler/BondSchemas/Execution/Execution.cs [hunk 5]
```diff
@@ -141,68 +136,79 @@ public partial class ArgumentValue
...
-    public partial class Argument
+    public partial class EntryPointOperation
     {
         [global::Bond.Id(5)]
-        public DataType Type { get; set; }
+        public string Name { get; set; }
 
         [global::Bond.Id(10)]
-        public string Name { get; set; }
+        public List<Argument> Arguments { get; set; }
```
Comment (ScottCarda-MS) [inline src/QsCompiler/BondSchemas/Execution/Execution.cs:160]: Along with my other comment, could this field be renamed as `Parameters`?

File: src/QsCompiler/BondSchemas/Execution/Execution.cs (hunk 4 above)
Comment (achocron) [reply src/QsCompiler/BondSchemas/Execution/Execution.cs:86]: Sure, that makes sense to me. I'll keep the name `ArgumentValue` as is, though, to make it very clear that it has a value (even though it is technically implicit)

Comment (ScottCarda-MS) [reply src/QsCompiler/BondSchemas/Execution/Execution.cs]: Thanks! That will make it easier for me to reason about this.

File: src/QsCompiler/Compiler/Compiler.csproj [hunk 6]
```diff
@@ -34,7 +38,7 @@
 
   <ItemGroup>
     <Compile Update="Templates\QirDriverCpp.cs">
-      <DesignTime>False</DesignTime>
+      <DesignTime>True</DesignTime>
```
Comment (cesarzc) [inline src/QsCompiler/Compiler/Compiler.csproj:41]: Can you please set this to `False`?

**Review by cesarzc — APPROVED** [review]

**Pull request merged**